### PR TITLE
[Fix] Don't @ mention when doing reply in DM

### PR DESCRIPTION
**Pull request by geekgonecrazy**
@RocketChat/core 

Previously doing a reply in a DM mentioned them.  Which is pointless.  You don't need to mention the person you're sending the message directly to :)

There might be a better way of determining if the room type is DM.  If so please suggest :)

### CODE REVIEW

File: packages/rocketchat-lib/client/MessageAction.js [hunk 1]
```diff
@@ -103,7 +103,12 @@ Meteor.startup(function() {
 			const message = this._arguments[1];
 			const input = instance.find('.input-message');
 			const url = RocketChat.MessageAction.getPermaLink(message._id);
-			const text = `[ ](${ url }) @${ message.u.username } `;
+			let text = `[ ](${ url }) `;
+
+			if (Session.get('openedRoom').indexOf(Meteor.userId()) === -1) {
+				text += `@${ message.u.username }`;
```
Comment (sampaiodiego) [inline packages/rocketchat-lib/client/MessageAction.js]: you're not adding an additional space after `@mention` as the previous code did.

File: packages/rocketchat-lib/client/MessageAction.js [hunk 2]
```diff
@@ -103,7 +103,12 @@ Meteor.startup(function() {
 			const message = this._arguments[1];
 			const input = instance.find('.input-message');
 			const url = RocketChat.MessageAction.getPermaLink(message._id);
-			const text = `[ ](${ url }) @${ message.u.username } `;
+			let text = `[ ](${ url }) `;
+
+			if (Session.get('openedRoom').indexOf(Meteor.userId()) === -1) {
```
Comment (sampaiodiego) [inline packages/rocketchat-lib/client/MessageAction.js]: I think it's safe doing `RocketChat.models.Rooms.findOne(message.rid, { fields: { t: 1 } })` to get room's type =)

File: packages/rocketchat-lib/client/MessageAction.js (hunk 1 above)
Comment (geekgonecrazy) [inline packages/rocketchat-lib/client/MessageAction.js]: Oops!  I was even thinking about it, I think I went back in and cleaned it out by mistake :(

Fixed now :+1:

File: packages/rocketchat-lib/client/MessageAction.js (hunk 2 above)
Comment (geekgonecrazy) [inline packages/rocketchat-lib/client/MessageAction.js]: That's a much cleaner way of doing it.  Thanks!

**Comment by geekgonecrazy** [comment]
@sampaiodiego could you take another look?  :smile:

**Comment by sampaiodiego** [comment]
what if you reply to a message sent by yourself? :thinking:

**Comment by marceloschmidt** [comment]
I think auto-replies should also remove @ my usernmae

**Comment by geekgonecrazy** [comment]
@sampaiodiego @marceloschmidt ok if you reply to self now doesn't mention :)

**Pull request merged**